src/frameworks/next/index.ts

```diff
@@ -390,6 +391,22 @@ export async function ɵcodegenFunctionsDirectory(sourceDir: string, destDir: st
  * Create a dev server.
  */
 export async function getDevModeHandle(dir: string, hostingEmulatorInfo?: EmulatorInfo) {
+  // throw error when using Next.js middleware with firebase serve
+  if (!hostingEmulatorInfo) {
+    const [middlewareJs, middlewareTs] = await Promise.all([
+      pathExists(join(dir, "middleware.js")),
+      pathExists(join(dir, "middleware.ts")),
+    ]);
+
+    if (middlewareJs || middlewareTs) {
+      throw new FirebaseError(
+        `${clc.bold("firebase serve")} does not support Next.js Middleware. Please use ${clc.bold(
+          "firebase emulators:start"
+        )} instead.`
+      );
+    }
```

Comment:
@jamesdaniels had some issues trying to import those files to check the exports to know whether we should error or not, so for now it's just checking if some of the middleware files exists

Reply:
can we add a dev/prod switch to isUsingMiddleware?